tests/test_client.py

```diff
@@ -0,0 +1,143 @@
+import unittest
+from unittest.mock import mock_open, patch, call
+from client.client import VariantAnnotationClient, EnsemblError
+from tests import util
+
+TEST_FILENAME = "variants.txt"
+
+TEST_URL = "https://test.com/{variant}"
+
+class TestVariantAnnotationClient(unittest.TestCase):
+    def test_parse_variants(self):
+        # This test string includes whitespace
+        TEST_FILE = '''variant1
+variant2    
+variant3'''
+        m = mock_open(read_data=TEST_FILE)
+        client = VariantAnnotationClient(TEST_URL)
+        with patch('builtins.open', m, create=True):
+            variants = client.parse_variants(TEST_FILENAME)
+            self.assertEqual(variants, [
+                "variant1",
+                "variant2",
+                "variant3",
+            ])
+        m.assert_called_once_with("variants.txt", "r")
+
+    def test_parse_variants_with_duplicates(self):
+        TEST_FILE = '''variant1
+variant1
+variant3'''
+        m = mock_open(read_data=TEST_FILE)
+        client = VariantAnnotationClient(TEST_URL)
+        with patch('builtins.open', m, create=True):
+            variants = client.parse_variants(TEST_FILENAME)
+            self.assertEqual(variants, [
+                "variant1",
+                "variant3"
+            ])
+        m.assert_called_once_with("variants.txt", "r")
+
+    def test_annotate_variant(self):
+        mock_request_dict = {
+            "https://test.com/variant1": {
+                "json": [
+                    {
+                        "assembly_name": "assembly_name",
+                        "seq_region_name": "seq_region_name",
+                        "start": "start",
+                        "end": "end",
+                        "most_severe_consequence": "most_severe_consequence",
+                        "strand": "strand",
+                        "transcript_consequences": [
+                            {
+                                "gene_symbol": "gene_symbol"
+                            }
+                        ]
+                    }
+                ],
+                "status_code": 200,
+            }
+        }
+        client = VariantAnnotationClient(TEST_URL)
```

Comment:
this could be setup as a [pytest fixture](https://docs.pytest.org/en/6.2.x/fixture.html) at the top of this file or in a `conftest.py` file. Rather than defining it in each test